src/main/java/com/betadevels/onlineshopping/action/subscription/CreateSubscriptionsAction.java

```diff
@@ -93,16 +95,22 @@ public SubscriptionListResponse invoke()
 					                                        .billingAddress( billingAddress )
 					                                        .shippingAddress( shippingAddress )
 					                                        .build();
+					for(Subscription s:userSubscriptions)
```

Comment:
Same thing here as well. We dont need to fetch the records to check for duplication. The db already has constraints for it. Just handle the db exceptions and map the exception messages to custom ones.